client-app/core/api/graphql/graphql-client.ts

```diff
@@ -1,11 +1,25 @@
 import { InMemoryCache } from "apollo-cache-inmemory";
 import { ApolloClient } from "apollo-client";
 import { HttpLink } from "apollo-link-http";
+import { onError } from "apollo-link-error";
+import _ from "lodash";
+
 const fetchPolicy = "no-cache";
 
+const httpLink = new HttpLink({ uri: `/xapi/graphql` });
+
+const errorHandler = onError(({ graphQLErrors }) => {
+  if (graphQLErrors && graphQLErrors.length) {
+    const unauthorizedError = _.find(graphQLErrors, (x) => x.extensions?.code === "Unauthorized");
+    if (unauthorizedError) {
+      location.href = "/sign-in";
```

Comment:
`const { hash, pathname, search } = location;`
`/sign-in?return=${pathname + search + hash}`